Dockerfile

```diff
@@ -3,13 +3,20 @@ LABEL authors="Drew Davies" \
       description="Docker image containing all software requirements for the nf-core/illuminavariantcalling pipeline"
 
 # Install the conda environment
-COPY environment.yml /
+COPY environment.yml  /
+RUN mkdir /data
+RUN cat environment.yml
```

Comment:
Yeah. I was just debugging. Couldn't find the source of the bug. The error ended up being from a memory error due to limitations on the docker container memory allocation limit.

Reply:
`docker build` seems to complete just fine even when you don't specify the memory limit or unlimited swap (see https://travis-ci.org/github/daviesdrew/variantcalling/builds/706281037#L200).

Your issue seems to be with the Snippy process (https://travis-ci.org/github/daviesdrew/variantcalling/builds/706658912#L3078) which you can fix by specifying the amount of memory for the Snippy process in your `conf/base.config` using 
```nextflow
withName: SNIPPY {
cpus = { check_max(4 * task.attempt, 'cpus') }
memory = { check_max( 8.GB * task.attempt, 'memory') }
}
```
or something like that. `check_max` is a function within your `nextflow.config`:
https://github.com/daviesdrew/variantcalling/blob/b8b730349b7ea9a1723bed488f259f0fbf625735/nextflow.config#L111-L140

Reply:
Okay, I will take out the memory swap of the docker build but I managed to remove the error with the --ram argument but it is not dynamic so I will change the base.config file to reflect available resources to the user.

Reply:
For grabbing the memory allocated to the task in gigabytes, you could try:

```nextflow
"""
snippy --cpus ${task.cpus} --ram ${task.memory.toGiga()} \\ 
"""
```

instead of https://github.com/daviesdrew/variantcalling/blob/a5ce48f1a2f36a07697cf71c67be8c8e1ab17aaa/modules/consensus.nf#L31-L32

Reply:
Alright I implemented this and it worked. Removed any memory args from the docker build and was able to remove the --ram argument from snippy process itself. 

I will be doing this for the rest of the processes. Maybe I can improve the performance of some processes.